File: app/src/main/java/com/hover/stax/balances/BalanceCardStackAdapter.java
package com.hover.stax.balances;

import android.content.Context;
import android.view.ViewGroup;

import androidx.cardview.widget.CardView;

import com.hover.stax.channels.Channel;
import com.hover.stax.databinding.StackBalanceCardBinding;
import com.hover.stax.utils.UIHelper;
import com.hover.stax.views.staxcardstack.StaxCardStackAdapter;
import com.hover.stax.views.staxcardstack.StaxCardStackView;

class BalanceCardStackAdapter extends StaxCardStackAdapter<Channel> {

    public BalanceCardStackAdapter(Context context) {
        super(context);
    }

    @Override
    public void bindView(Channel channel, int pos, StaxCardStackView.ViewHolder holder) {
        if(holder instanceof MyViewHolder) {
            String hex = channel.primaryColorHex;
            ((MyViewHolder) holder).onBind(hex);
        }
    }

    @Override
    protected StaxCardStackView.ViewHolder onCreateView(ViewGroup parent, int viewType) {
        return new MyViewHolder(StackBalanceCardBinding.inflate(getLayoutInflater(), parent, false));
    }

    static class MyViewHolder extends StaxCardStackView.ViewHolder {
        private final CardView cardView;
        public MyViewHolder(StackBalanceCardBinding binding) {
            super(binding.getRoot());
            cardView = binding.getRoot();
        }
        public void onBind(String hex) {
            cardView.setCardBackgroundColor(UIHelper.getColor(hex, false, getContext()));
        }
    }

}

File: app/src/main/java/com/hover/stax/balances/BalanceCardStackAdapter.kt
package com.hover.stax.balances

import android.content.Context
import android.view.ViewGroup
import com.hover.stax.account.Account
import com.hover.stax.channels.Channel
import com.hover.stax.databinding.StackBalanceCardBinding
import com.hover.stax.utils.UIHelper
import com.hover.stax.views.staxcardstack.StaxCardStackAdapter
import com.hover.stax.views.staxcardstack.StaxCardStackView

class BalanceCardStackAdapter(private val ctx: Context): StaxCardStackAdapter<Account>(ctx) {

    override fun onCreateView(parent: ViewGroup?, viewType: Int): StaxCardStackView.ViewHolder {
        return MyViewHolder(StackBalanceCardBinding.inflate(layoutInflater, parent, false))
    }

    override fun bindView(account: Account, position: Int, holder: StaxCardStackView.ViewHolder?) {
        if(holder is MyViewHolder) holder.bind(account.primaryColorHex)
    }

    inner class MyViewHolder(val binding: StackBalanceCardBinding): StaxCardStackView.ViewHolder(binding.root) {

        fun bind(hex: String) {
            binding.root.setCardBackgroundColor(UIHelper.getColor(hex, false, ctx))
        }
    }
}